File: Stack/StackImplementation.cpp
// Stack follows LIFO(Last In First Out)
#include <iostream>
using namespace std;
#define MAX_SIZE 1000

class Stack{
    public:
        int top = -1;
        int keepStack[MAX_SIZE];

    bool isEmpty(){
        return (top == -1) ? true : false;
    }

    bool isFull(){
        return top >= MAX_SIZE ? true : false;
    }

    int size(){
        return top + 1;
    }

    void push(int data){
        if (isFull()){
            cout << "Stack is Full " << endl;
            return;
        }
        keepStack[++top] = data;
    }

    int pop(){
        if (isEmpty()){
            cout << "Stack is empty you cant pop " << endl;
        }
        return keepStack[top--];
    }
};

int main(){

    Stack *mystack = new Stack();

    mystack->push(2);
    mystack->push(3);
    mystack->push(4);

    cout << "Stack is Empty ? : " << (mystack->isEmpty() ? "yes" : "no") << endl;
    cout << "Stack is Full ?  : " << (mystack->isFull() ? "yes" : "no") << endl;
    int popingItem = mystack->pop();
    cout << "Poping item : " << popingItem << endl;
    cout << "Stack Size : " << mystack->size() << endl;

    return 0;
}

File: Stack/StackImplementation.java
public class StackImplementation {

    static class Stack {

        private int MAX_SIZE = 1000;
        private int top = -1;
        public int[] keepStack = new int[MAX_SIZE];

        public boolean isEmpty() {
            return top == -1 ? true : false;
        }

        public boolean isFull() {
            return top >= MAX_SIZE ? true : false;
        }

        public int size() {
            return top + 1;
        }

        public void push(int data) {
            if (isFull()) {
                System.out.println("Stack is full");
                return;
            }
            keepStack[++top] = data;
        }

        public int pop() {
            if (isEmpty()) {
                System.out.println("Stack is empty. Can't pop .");
            }
            return keepStack[top--];
        }
    }

    public static void main(String[] args) {

        Stack myStack = new Stack();

        myStack.push(2);
        myStack.push(3);
        myStack.push(4);

        System.out.println("Is Stack Empty ? :" + myStack.isEmpty());
        System.out.println("Is Stack Full ? :" + myStack.isFull());
        System.out.println("Stack size :  :" + myStack.size());
        int popingItem = myStack.pop();
        System.out.println("Poping Item :" + popingItem);
        System.out.println("Now Vector size  :" + myStack.size());
    }
}
